src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -237,6 +255,17 @@ class FileField extends React.Component {
         ? uiSchema['ui:title']
         : schema.title;
 
+    const allowUploadWhenError =
+      !enableShortWorkflow ||
+      (enableShortWorkflow &&
+        !files.some((file, index) => {
+          const errors =
+            _.get([index, '__errors'], errorSchema) ||
+            [file.errorMessage].filter(error => error);
+
+          return errors.length > 0;
+        }));
```

Comment:
Does the condition after the `||` mean that this would allow uploads when there _aren't_ errors? I see the `!` has been added to the beginning of `files`.

Reply:
Yup. If the flag is on, only allow additional uploads when there are no errors present.

Reply:
Does the `WhenError` part of the name still make sense?

Reply:
Because the feature flag is mixed in there, it does. The variable will need a rename when the flag is removed. With the flag off (`!enableShortWorkflow`) we `allowUploadWhenError`. When the flag is on, we don't `allowUploadWhenError`.

Reply:
I think this should be renamed to reflect what the value represents regardless of whether or not the feature flag is flipped (`allowUpload` possibly), and a comment added explaining how the feature flag changes the criteria for allowing an upload.

This is defined on line 258 and not used until line 450. If a future person is reading this code and sees `allowUploadWhenError && ...`, there's a good chance that they may not be thinking "but invert this if a specific feature flag is on", because the definition won't be visible. Even if the feature flag is removed and becomes the default a week from now, I'd rather avoid some potential awkwardness with the naming.

Reply:
Ok, finally understood what you were telling me. Name changed and comment added. And no worries on not making staging review. I don't think anyone on my team knew we had to get the artifact in staging a week before the staging review date. We all thought it was by the date of the staging review. 

And I really appreciate your thoroughness, especially since I am very new to va.gov.